service/plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/route/handler/base/DataAggregationRouteBuilder.java

```diff
@@ -86,7 +87,8 @@ public void configure() throws Exception {
                 .log(LoggingLevel.TRACE, "agg complete ${body}")
                 .process(processor)
                 // Checking for empty scan types.
-                .choice().when(simple("${body.size} == 0"))
+                .choice().when(and(simple("${body.size} == 0"),
+                        header(RouteConstants.DATA_PROCESSING_COMPLETE).isEqualTo(true)))
```

Comment:
This is the main change, the rest are just to improve the code. Here I added an and clause that checks that we only mark the scan types as complete when we are processing the final batch of data (DATA_PROCESSING_COMPLETE is set to true).